tests/functional/aws-node-sdk/test/object/listParts.js

```diff
@@ -0,0 +1,77 @@
+import assert from 'assert';
+
+import withV4 from '../support/withV4';
+import BucketUtility from '../../lib/utility/bucket-util';
+
+const bucket = 'bucketlistparts';
+const key = 'key';
+const bodyFirstPart = Buffer.allocUnsafe(10);
+const bodySecondPart = Buffer.allocUnsafe(20);
+
+function checkNoError(err) {
+    assert.equal(err, null,
+        `Expected success, got error ${JSON.stringify(err)}`);
+}
+
+describe('List parts', () => {
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+        let uploadId;
+        let secondEtag;
+
+        beforeEach(() => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return s3.createBucketAsync({ Bucket: bucket })
+            .then(() => s3.createMultipartUploadAsync({
+                Bucket: bucket, Key: key }))
+            .then(res => {
+                uploadId = res.UploadId;
+                const params = { Bucket: bucket, Key: key, PartNumber: 1,
+                    UploadId: uploadId, Body: bodyFirstPart };
+                return s3.uploadPartAsync(params);
```

Comment:
For the sake of consistency, you might pass the params object directly into `s3.uploadPartAsync` as you've done just below for part 2.